src/dhcp_service.rs

```diff
@@ -67,11 +95,21 @@ impl DhcpService {
         ))
     }
 
-    pub fn release_lease(mut self) -> Result<(), DhcpServiceError> {
+    pub fn release_lease(
+        mut self,
+        siaddr: IpAddr,
+        yiaddr: IpAddr,
+        srv_addr: IpAddr,
```

Comment:
wouldn't it make more sense to just pass the lease as argument and then correctly convert it into an ipv4/v6 address inside the match block?

the current conversion logic feels wrong, you just silently ignore the errors and default to a 0 address?